adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -75,6 +75,11 @@ public void setMessageMapping(final MessageMapping messageMapping) {
 
     private Future<MappedMessage> mapMessage(final MqttContext ctx, final ResourceIdentifier targetAddress,
                                  final MqttPublishMessage message) {
+        if (messageMapping == null) {
```

Comment:
is the *MessageMapping* bean required or not? In the Confiugration you create the bean so it will get injected. The *setMessageMapping* method's `@Autowired` annotation also requires a mapping to be injected (otherwise it should be `@Autowired(required = false)`. That setter should also throw a NullPointerException if the given mapping is `null` ...